terraform/crates-io/compute-static/src/main.rs

```diff
@@ -0,0 +1,60 @@
+use fastly::http::{Method, StatusCode};
+use fastly::{Error, Request, Response};
+use log::warn;
+
+use crate::config::Config;
+
+mod config;
+
+#[fastly::main]
+fn main(request: Request) -> Result<Response, Error> {
+    let config = Config::from_dictionary();
+
+    if let Some(response) = limit_http_methods(&request) {
+        return Ok(response);
+    }
+
+    send_request_to_s3(&config, &request)
+}
+
+/// Limit HTTP methods
+///
+/// Clients are only allowed to request resources using GET and HEAD requests. If any other HTTP
+/// method is received, HTTP 403 Unauthorized is returned.
+///
+/// We don't return HTTP 405 Method Not Allowed to maintain parity with CloudFront.
+fn limit_http_methods(request: &Request) -> Option<Response> {
+    let method = request.get_method();
+
+    if method != Method::GET && method != Method::HEAD {
+        return Some(
+            Response::from_body("Method not allowed").with_status(StatusCode::UNAUTHORIZED),
+        );
+    }
+
+    None
+}
+
+/// Forward client request to S3
+///
+/// The request that was received by the client is forwarded to S3. First, the primary bucket is
+/// queried. If the response indicates a server issue (status code >= 500), the request is sent to
+/// a fallback bucket in a different geographical region.
+fn send_request_to_s3(config: &Config, request: &Request) -> Result<Response, Error> {
+    let primary_request = request.clone_without_body();
+
+    let mut response = primary_request.send(&config.primary_host)?;
+    let status_code = response.get_status().as_u16();
+
+    if status_code >= 500 {
+        warn!(
```

Comment:
The setup for [logging](https://developer.fastly.com/learning/compute/rust/#logging) is not complete yet, and we are not [streaming logs](https://docs.fastly.com/en/guides/about-fastlys-realtime-log-streaming-features). So this warning will not show up anywhere.